add-custom-dynamic-entity-data-source/src/main/java/com/esri/arcgismaps/sample/addcustomdynamicentitydatasource/components/CustomEntityFeedProvider.kt

```diff
@@ -0,0 +1,172 @@
+package com.esri.arcgismaps.sample.addcustomdynamicentitydatasource.components
+
+import com.arcgismaps.data.Field
+import com.arcgismaps.data.FieldType
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.realtime.CustomDynamicEntityDataSource
+import com.arcgismaps.realtime.DynamicEntityDataSourceInfo
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.NonCancellable
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.double
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import java.io.File
+import java.io.IOException
+
+class CustomEntityFeedProvider(
+    fileName: String,
+    private val entityIdField: String,
+    private val delayDuration: Long
```

Comment:
I suggest to use `kotlin.time.Duration` here, which allows the user to use any time unit they like. Also not quite clear what this delay is used for. Some property doc would help.
```suggestion
    private val delayDuration: Duration
```

Then user code could be:
```kotlin
private val customSource = CustomEntityFeedProvider(
        "$provisionPath/AIS_MarineCadastre_SelectedVessels_CustomDataSource.jsonl",
        "MMSI",
        10.milliseconds
    )
```